src/file_organizer/web/settings_routes.py

```
@@ -196,6 +197,48 @@ def _render_section(
+    """Run the shared action → render-success / render-error flow for a section POST.
+
+    ``action`` performs the persistence (usually a ``settings_service.apply_*``
+    call) and returns the updated ``(WebSettings, AppConfig)`` pair to render.
+    A ``ValueError`` is treated as a validation rejection and routed to
+    ``on_rejected`` when provided (which owns its own logging and rendering);
+    any other failure logs ``failure_log`` and re-renders the section from
+    freshly loaded state with an error flash.
+    """
+    try:
+        ws, app_config = action()
+    except Exception as exc:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)